Controller/Callback/Offsite.php

```diff
@@ -84,7 +85,7 @@ public function execute()
         }
         
         $paymentMethod = $payment->getMethod();
-        if (! in_array($paymentMethod, [Alipay::CODE, Internetbanking::CODE, Installment::CODE, Truemoney::CODE])) {
+        if (! in_array($paymentMethod, [Alipay::CODE, Internetbanking::CODE, Installment::CODE, Truemoney::CODE, Citipoints::CODE])) {
```

Comment:
Feels like this array should be named and defined elsewhere - maybe `validPaymentMethods` or something

Reply:
Also, this `execute` function is HUGE - maybe we should break it up?

Reply:
how would you break it up?

I don't hate that array defined here, what @guzzilar thinks about that?

Reply:
also:
Any refactoring changes that are not consistent with any file, might be done in a separated file.

Reply:
IIRC, I think this topic has been discussed several times and I agreed that this method is getting big and would be nice if we can lean it down a bit to avoid any potential one error break all payments.

If you ask me, as I don't have a particular idea on how to do it, so I'm ok with what it is for now. Maybe I'll also submit an alternative refactoring idea next year on Jan or Feb 😉